xwiki-platform-core/xwiki-platform-mentions/xwiki-platform-mentions-api/src/main/java/org/xwiki/mentions/MentionNotificationService.java

```diff
@@ -42,6 +42,6 @@
      * @param location The location of the mention.
      * @param anchorId The anchor link to use.
      */
-    void sendNotif(DocumentReference authorReference, DocumentReference documentReference,
-        DocumentReference mentionedIdentity, MentionLocation location, String anchorId);
+    void sendNotif(String authorReference, String documentReference, DocumentReference mentionedIdentity,
```

Comment:
Since you refactor it, I would have renamed it `sendNotification`: no real need to use an abbreviation there...
Now I'm not sure about this refactoring: what's the need for not using a real reference type there? (looking in order so I'll probably get the answer below)